gandi/cli/modules/paas.py

```diff
@@ -29,6 +32,86 @@ def type_list(cls, options=None):
         return cls.safe_call('paas.type.list', options)
 
     @classmethod
+    def clone(cls, name, vhost, directory):
+        """Clone a PaaS instance's vhost into a local git repository."""
+        paas_info = cls.info(name)
+
+        paas_access = '%s@%s' % (paas_info['user'], paas_info['git_server'])
+
+        remote_url = 'ssh+git://%s/%s.git' % (paas_access, vhost)
+
+        init_git = cls.execute('git clone %s %s'
+                               % (remote_url, directory))
+        if init_git:
+            return cls.save_config(paas_info, paas_access, vhost, directory)
+
+    @classmethod
+    def attach(cls, name, vhost, remote_name):
+        """Attach an instance's vhost to a remote from the local repository."""
+        paas_access = cls.get('paas_access')
+
+        if not paas_access:
+            paas_info = cls.info(name)
+            paas_access = '%s@%s' % (paas_info['user'], paas_info['git_server'])
+
+        remote_url = 'ssh+git://%s/%s.git' % (paas_access, vhost)
+
+        ret = cls.execute('git remote add %s %s' % (remote_name, remote_url,))
+
+        if ret:
+            cls.echo('Adding remote `%s` to your local git repository.'
+                     % (remote_name))
+            cls.echo('Use `git push %s master` to push your code to the '
+                     'instance.' % (remote_name))
+            cls.echo('Then `$ gandi deploy` to build and deploy your '
+                     'application.')
+
+            return cls.save_config(paas_info, paas_access, vhost,)
+
+    @classmethod
+    def deploy(cls):
+        paas_access = cls.get('paas.access')
+        deploy_git_host = cls.get('paas.deploy_git_host')
+
+        if not paas_access:
+            cls.error("Deploy requires a local configuration file.")
+            return
+
+        cls.execute("ssh %s 'deploy %s'" % (paas_access, deploy_git_host,))
+
+    @classmethod
+    def clone(cls, name, vhost, directory):
+        """Clone a PaaS instance's vhost into a local git repository."""
+        paas_info = cls.info(name)
+
+        git_server = paas_info['git_server']
+        user = paas_info['user']
+
+        paas_access = '%s@%s' % (user, git_server)
+        git_remote = 'ssh+git://%s/%s.git' % (paas_access, vhost)
```

Comment:
I wonder if it will be interesting to do a (private) function to do this (you do it few time and not in the same way).
Maybe cache the call to paas_info also (with an other private method)?